### PR TITLE
Add the XColor type structure

### PR DESCRIPTION
**Pull request by gabbapeople**
Add XColor type

### CODE REVIEW

File: workspace/__lib__/xod/color/color-hsl/patch.cpp [hunk 1]
```diff
@@ -1,42 +1,53 @@
 
-struct State {};
+using xod::XColor::XColor;
```
Comment (nkrkv) [inline workspace/__lib__/xod/color/color-hsl/patch.cpp]: Why using `using` here? Should not be required.

File: workspace/__lib__/xod/color/color-hsl/patch.cpp [hunk 2]
```diff
@@ -1,42 +1,53 @@
...
+
+struct State {
+    XColor color = { 0, 0, 0 };
+};
```
Comment (nkrkv) [inline workspace/__lib__/xod/color/color-hsl/patch.cpp]: This should be stateless, as it was.

File: workspace/__lib__/xod/color/color-hsl/patch.cpp [hunk 3]
```diff
@@ -1,42 +1,53 @@
...
-  if ((2.0f * vH) < 1.0f) return v2;
+    if ((2.0f * vH) < 1.0f)
+        return v2;
 
-  if ((3.0f * vH) < 2.0f) return (v1 + (v2 - v1) * ((2.0f / 3.0f) - vH) * 6.0f);
+    if ((3.0f * vH) < 2.0f)
+        return (v1 + (v2 - v1) * ((2.0f / 3.0f) - vH) * 6.0f);
 
-  return v1;
+    return v1;
```
Comment (nkrkv) [inline workspace/__lib__/xod/color/color-hsl/patch.cpp]: Please, extract all `clang-format` boilerplate to a single, separate commit.

File: workspace/__lib__/xod/color/color-rgb-bytes/patch.cpp [hunk 4]
```diff
@@ -1,13 +1,20 @@
+using xod::XColor;
 
 struct State {
+    XColor color = { 0, 0, 0 };
```
Comment (nkrkv) [inline workspace/__lib__/xod/color/color-rgb-bytes/patch.cpp]: ditto

File: workspace/__lib__/xod/color/color/patch.cpp [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
-};
+// clang-format off
+{{#global}}
+// clang-format on
+
+#include <XColor.h>
+
+// clang-format off
+{{/global}}
+// clang-format on
```
Comment (nkrkv) [inline workspace/__lib__/xod/color/color/patch.cpp]: Should not be here. `Color` is available out of the box.

File: workspace/__lib__/xod/math/sin/patch.xodp [hunk 6]
```diff
@@ -5,17 +5,17 @@
       "id": "Bkpwy22I-",
       "position": {
         "units": "slots",
-        "x": 0.25,
-        "y": 2.15
+        "x": 0,
+        "y": 2
```
Comment (nkrkv) [inline workspace/__lib__/xod/math/sin/patch.xodp]: Out of this PR scope.

**Comment by nkrkv** [comment]
```
Conflicting files
packages/xod-arduino/platform/memory.h
packages/xod-arduino/platform/types.h
```

You should rebase the branch before the merge

**Comment by gabbapeople** [comment]
@nkrkv added just the new type.

File: packages/xod-arduino/platform/types.h [hunk 7]
```diff
@@ -14,10 +14,12 @@ typedef double Number;
 typedef bool Logic;
 typedef unsigned long TimeMs;
 typedef uint8_t ErrorFlags;
-
 struct Pulse {
```
Comment (nkrkv) [inline packages/xod-arduino/platform/types.h]: Do not remove the splitting empty line

File: packages/xod-arduino/platform/types.h [hunk 8]
```diff
@@ -14,10 +14,12 @@ typedef double Number;
...
   Pulse() {}
   Pulse(bool) {}
   Pulse(int) {}
 };
+struct XColor {
+  uint8_t r, g, b;
+};
```
Comment (nkrkv) [inline packages/xod-arduino/platform/types.h]: Empty lines before and after the definition are missing

File: workspace/__lib__/xod/color/color/patch.cpp [hunk 9]
```diff
@@ -2,9 +2,7 @@
 struct State {
 };
 
-struct Type {
-    uint8_t r, g, b;
-};
+using Type = xod::XColor;
```
Comment (nkrkv) [inline workspace/__lib__/xod/color/color/patch.cpp]: You’re already in the `xod` namespace.

```suggestion
using Type = XColor;
```

**Pull request merged**